### PR TITLE
Document releasing

### PR DESCRIPTION
**Pull request by chmeliik**
STONEBLD-1270

# Maintainers will complete the following section

- [x] Commit messages are descriptive enough
- [n/a] Code coverage from testing does not decrease and new code is covered
- [x] Docs updated (if applicable)
- [n/a] Docs links in the code are still valid (if docs were updated)

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -250,6 +251,23 @@ Generate data for test cases matching a pytest pattern:
...
 ```
 
+## Releasing
+
+To release a new version of Cachi2, simply create a [GitHub release][cachi2-releases]. Note that
+Cachi2 follows [semantic versioning](https://semver.org/) rules.
+
+Upon release, the [.tekton/release.yaml](.tekton/release.yaml) pipeline tags the corresponding
+[Cachi2 image][cachi2-container] with the newly released version tag (after validating that the
+tag follows the expected format: `$major.$minor.$patch`, without a `v` prefix).
```
[the remark this answers is not in the document]
Comment (ben-alkov) [reply README.md:261]: I think that as long as someone has read from the top and seen the "Container image" section above, it should be clear...

[the remark this answers is not in the document]
Comment (chmeliik) [reply README.md:261]: I added an explanation anyway :smile:

**Review by ejegrova — APPROVED** [review]
LGTM

**Review by lkolacek — APPROVED** [review]
LGTM :+1:

**Pull request merged**